src/test/java/org/jabref/gui/search/GlobalSearchBarTest.java

```diff
@@ -0,0 +1,156 @@
+package org.jabref.gui.search;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+
+import javafx.scene.Scene;
+import javafx.scene.control.TextInputControl;
+import javafx.scene.layout.HBox;
+import javafx.stage.Stage;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.JabRefFrame;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.undo.CountingUndoManager;
+import org.jabref.gui.util.DefaultTaskExecutor;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.search.rules.SearchRules;
+import org.jabref.preferences.PreferencesService;
+import org.jabref.preferences.SearchPreferences;
+import org.jabref.testutils.category.GUITest;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Answers;
+import org.testfx.api.FxRobot;
+import org.testfx.framework.junit5.ApplicationExtension;
+import org.testfx.framework.junit5.Start;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * The GlobalSearchBarTest class currently serves the GUI testing regarding
+ * search history recording improvement. Designed in TestFX testing framework
+ * and using FxRobot utility class, in order to automate the ui testing, by
+ * simulating interactions between the user and the GUI (i.e. click on
+ * a node, write to a text input control etc.). The tests included verify
+ * that neither redundant data are recorded to the search history nor
+ * empty search queries are stored.
+ *
+ * @author dkokkotas
+ * @version 1.0
+ * @since May 2023
+ */
+
+@GUITest
+@ExtendWith(ApplicationExtension.class)
+public class GlobalSearchBarTest {
+    private Stage stage;
+    private Scene scene;
+    private HBox hBox;
+
+    private GlobalSearchBar searchBar;
+    private StateManager stateManager;
+
+    /**
+     * The onStart method, accepts the stage of the application and sets
+     * up the basic components of the UI (i.e. scene, layout), that will
+     * be used for our tests. Also interacts with GlobalSearchBar and
+     * StateManager classes, in order to set up the search box.
+     *
+     * @param stage Reference to the Stage class, part of the JavaFX UI toolkit
+     */
```

Comment:
May I ask if this comment is really necessary? Doesn't one see it directly from the code?

Reason of minimizing comments: If the code is updated, sometimes the JavaDoc is not updated, and then the whole thing diverges, which leads to higher effort for maintenance.

Reply:
> May I ask if this comment is really necessary? Doesn't one see it directly from the code?

I just considered an explanatory documentation both on class and it's methods to be appropriate. However, if a detailed approach is redundant and causes higher effort for maintenance, it must be removed or shortened.